packages/lens/src/validate.ts

```diff
@@ -20,53 +21,96 @@ export async function hasAddressCollectedPost(
   if (collectAddress === null) {
     return false
   }
-  const amountOwned = await checkAddressOwnsCollect(address, collectAddress)
-  return amountOwned > 0n
+  return await checkAddressMintedCollect(address, collectAddress)
 }
 
 export async function getCollectAddress(postId: string) {
-  const result = await lensClient.publication.fetch({
-    forId: postId,
-  })
-  if (!result || result?.__typename === 'Mirror') {
+  try {
+    const result = await lensClient.publication.fetch({
+      forId: postId,
+    })
+    if (!result || result?.__typename === 'Mirror') {
+      return null
+    }
+    const openActions = result.openActionModules
+    if (!openActions || openActions.length === 0) {
+      return null
+    }
+    const collectActions = openActions.filter(
+      (action) =>
+        action.__typename === 'SimpleCollectOpenActionSettings' ||
+        action.__typename === 'MultirecipientFeeCollectOpenActionSettings',
+    ) as Array<
+      | MultirecipientFeeCollectOpenActionSettingsFragment
+      | SimpleCollectOpenActionSettingsFragment
+    >
+
+    const collectNft = collectActions.find((action) => action.collectNft)
+    const collectAddress = collectNft?.collectNft?.toLowerCase()
+    return (collectAddress as Address) ?? null
+  } catch (error) {
+    console.error('Error fetching collect contract address', error)
     return null
   }
-  const openActions = result.openActionModules
-  if (!openActions || openActions.length === 0) {
-    return null
+}
+
+export async function checkAddressMintedCollect(
+  actor: Address,
+  collectAddress: Address,
+) {
+  try {
+    return await checkMintedUsingAlchemy(actor, collectAddress)
+  } catch (err) {
+    console.error('Error while using alchemy api', err)
+    return await checkMintedUsingReservoir(actor, collectAddress)
   }
-  const collectActions = openActions.filter(
-    (action) =>
-      action.__typename === 'SimpleCollectOpenActionSettings' ||
-      action.__typename === 'MultirecipientFeeCollectOpenActionSettings',
-  ) as Array<
-    | MultirecipientFeeCollectOpenActionSettingsFragment
-    | SimpleCollectOpenActionSettingsFragment
-  >
+}
 
-  const collectNft = collectActions.find((action) => action.collectNft)
-  const collectAddress = collectNft?.collectNft?.toLowerCase()
-  return (collectAddress as Address) ?? null
+async function checkMintedUsingAlchemy(
+  actor: Address,
+  collectAddress: Address,
+) {
+  // alchemy will work without an api key, but we risk being rate-limited
+  if (!process.env.ALCHEMY_API_KEY) {
+    console.error('Alchemy API key not found')
+  }
+
+  const options = {
+    contractAddresses: [collectAddress],
+  }
+
+  const transfers = await alchemy.nft.getTransfersForOwner(
+    actor,
+    GetTransfersForOwnerTransferType.TO,
+    options,
+  )
+
+  // only count nfts that originate from zeroAddress
+  const mintedNfts = transfers.nfts.filter((nft) => nft.from === zeroAddress)
```

Comment:
Clever - you know we could actually handle this just with the contract `Transfer` events. Just ship this implementation though that's hilarious it's full circle though.